### PR TITLE
Rename files by date-time stamp and also provide other utilities

### PR DESCRIPTION
**Pull request by tanchifu**
I changed the code as described in the following points:

1. The files will be renamed to their date-time stamps found in the EXIF info. If the EXIF doesn’t exist in the file, the modified date timestamp will be used instead.
2. The target folder structure will not have “Day” sub-folder created. All pictures/videos of the same month will be placed in the “Month” sub-folder.
3. Only files with proper picture/video extension names (Ex. jpg, gif, png, mov, mp4, etc.) will be processed.
4. Added --backup option to copy files from source folder to the structured destination folder without renaming the files.
5. Added --report option to report total number of files and sizes. Also report the file counts by the camera models and file extensions.
6. Added -j <META-FILE> argument to output the meta info of the source files in json format to the <META-FILE>.
7. The --use-modif-date option is removed.

Example of running the --report option:

~/github/picture-sorter> ./picture-sorter -s ~/test --report
Reading top folder: ~/test
-----------------------------------
Total # of files: 9. Total size: 145.93 MB
   --------------------------------
   Camera -COOLPIX S9700 : 4
   Camera -SPH-L710 : 1
   Camera unknown : 4
   --------------------------------
   JPEG : 1
   JPG : 7
   M4V : 1
   --------------------------------

### CODE REVIEW

**Comment by jscappini** [comment]
@tanchifu Hi! The code and documentation looks really good. Let me do a quick revision tonight and I'll be merging the PR into master. Thanks!

File: picture-sorter [hunk 1]
```diff
@@ -1,15 +1,24 @@
...
-if (!(src && dst)) {
+if (!(src && (dst || reportOnly || metaFile))) {
   console.log("");
   console.log("Usage: picture-sorter options");
   console.log("");
-  console.log("   Ex: picture-sorter -m --use-modif-date -s ~/tmp -d ~/tmp/sorted");
+  console.log("   Ex 1: picture-sorter -m -s ~/tmp -d ~/tmp/sorted");
+  console.log("       This command line renames and moves the picture files to the destination folder.");
+  console.log("");
+  console.log("   Ex 2: picture-sorter -report -s ~/tmp");
```
Comment (jscappini) [inline picture-sorter]: There is a '-' missing here (--report instead of -report).

**Comment by jscappini** [comment]
@tanchifu It looks really good! Just correct that typo (see comments on the code) and I will merge it. Would you do me a favor? Could you please delete the file "index.coffee"? It is no longer necessary. Thanks.

**Comment by tanchifu** [comment]
Hi Javier,
Yes. Thanks for the review. I will work on it tonight.
Tanchi

**Comment by tanchifu** [comment]
I corrected the -report error and also removed the file index.coffee.

**Pull request merged**